### PR TITLE
Add configurableProperties to support app customization

### PR DESCRIPTION
**Pull request by vganiga**
Adding array of enums to support configurable app properties

### CODE REVIEW

File: MicrosoftTeams.schema.json [hunk 1]
```diff
@@ -805,6 +805,24 @@
       },
       "description": "When a group install scope is selected, this will define the default capability when the user installs the app",
       "additionalProperties": false
+    },
+    "configurableProperties": {
+      "type": "array",
+      "description": "A list of tenant configurable properties for an app",
+      "maxItems": 9,
```
Comment (asnow003) [inline MicrosoftTeams.schema.json:812]: What is this max number used for?

File: MicrosoftTeams.schema.json [hunk 2]
```diff
@@ -805,6 +805,24 @@
...
+      "items": {
+        "enum": [
+          "name",
+          "shortDescription",
+          "longDescription",
+          "smallImageUrl",
```
Comment (asnow003) [inline MicrosoftTeams.schema.json:818]: See the section below how they group the image urls to the grouping of "icons"  we should follow that same pattern.

File: MicrosoftTeams.schema.json [hunk 3]
```diff
@@ -805,6 +805,24 @@
...
+          "largeImageUrl",
+          "accentColor",
+          "developerUrl",
```
Comment (asnow003) [inline MicrosoftTeams.schema.json:821]: What does developerUrl map to?

File: MicrosoftTeams.schema.json [hunk 4]
```diff
@@ -805,6 +805,24 @@
       },
       "description": "When a group install scope is selected, this will define the default capability when the user installs the app",
       "additionalProperties": false
+    },
+    "configurableProperties": {
```
Comment (greybax) [inline MicrosoftTeams.schema.json:809]: @Niranjan-S seems we have to include these changes into `Microsoft.Teams.Apps.Schema` project as well

**Review by greybax — APPROVED** [review]

File: MicrosoftTeams.schema.json [hunk 5]
```diff
@@ -805,6 +805,24 @@
...
+      "type": "array",
+      "description": "A list of tenant configurable properties for an app",
+      "maxItems": 9,
+      "items": {
+        "enum": [
+          "name",
+          "shortDescription",
```
Comment (asnow003) [inline MicrosoftTeams.schema.json:816]: Same with required section below we are using the top level grouping... of "description"

**Review by asnow003 — CHANGES_REQUESTED** [review]

File: MicrosoftTeams.schema.json (hunk 4 above)
Comment (Niranjan-S) [reply MicrosoftTeams.schema.json:809]: @greybax Why we need these changes?

**Comment by Niranjan-S** [comment]
@vganiga  Can you please add a bit detailed description to this PR?

File: MicrosoftTeams.schema.json (hunk 1 above)
Comment (vganiga) [reply MicrosoftTeams.schema.json:812]: maxItems is the maximum number of items configurableProperties array can hold, which is all the enum values

Comment (vganiga) [reply MicrosoftTeams.schema.json:812]: > What is this max number used for?

maxItems is the maximum number of items configurableProperties array can hold, which is all the enum values

File: MicrosoftTeams.schema.json (hunk 3 above)
Comment (vganiga) [reply MicrosoftTeams.schema.json:821]: developerUrl maps to the developer.websiteUrl

File: MicrosoftTeams.schema.json (hunk 2 above)
Comment (vganiga) [reply MicrosoftTeams.schema.json:818]: > See the section below how they group the image urls to the grouping of "icons" we should follow that same pattern.

I just want the list of these enum to be in the array to indicate they are configurable, grouping will make it complicated

**Review by Niranjan-S — APPROVED** [review]

**Review by asnow003 — APPROVED** [review]

**Pull request merged**